packages/block-editor/src/store/reducer.js

```diff
@@ -264,11 +275,12 @@ const withBlockCache = ( reducer ) => ( state = {}, action ) => {
 			};
 			break;
 		case 'REPLACE_BLOCKS_AUGMENTED_WITH_CHILDREN':
+			const parentClientIds = fillKeysWithEmptyObject( getBlocksWithParentsClientIds( action.replacedClientIds ) );
+
 			newState.cache = {
 				...omit( newState.cache, action.replacedClientIds ),
-				...fillKeysWithEmptyObject(
-					getBlocksWithParentsClientIds( keys( flattenBlocks( action.blocks ) ) ),
-				),
+				...omit( parentClientIds, action.replacedClientIds ),
+				...fillKeysWithEmptyObject( keys( flattenBlocks( action.blocks ) ) ),
```

Comment:
To explain a bit what I've done here:
1. The `replacedClientIds` needs to be used to find parentClientIds, but then that clientId needs to be omitted, hence the second call to `omit` here.
2. The new clientId needs to be included in the cache, and there's a unit test `should replace the block even if the new block clientId is the same`, so that's why it's the last item is adding the new client id.

Reply:
The logic seems good to me.